### PR TITLE
feat(types): improve Proposal and SynchronyParams validation tests

### PR DESCRIPTION
**Pull request by cason**
Contributes to #1731

---

#### PR checklist

- [x] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: types/canonical.go [hunk 1]
```diff
@@ -42,9 +42,9 @@ func CanonicalizePartSetHeader(psh cmtproto.PartSetHeader) cmtproto.CanonicalPar
 func CanonicalizeProposal(chainID string, proposal *cmtproto.Proposal) cmtproto.CanonicalProposal {
 	return cmtproto.CanonicalProposal{
 		Type:      ProposalType,
-		Height:    proposal.Height,       // encoded as sfixed64
-		Round:     int64(proposal.Round), // encoded as sfixed64
-		POLRound:  int64(proposal.PolRound),
+		Height:    proposal.Height,          // encoded as sfixed64
+		Round:     int64(proposal.Round),    // encoded as sfixed64
+		POLRound:  int64(proposal.PolRound), // FIXME: not matching
```
Comment (sergio-mena) [inline types/canonical.go:47]: ???

File: types/params.go [hunk 2]
```diff
@@ -247,11 +251,11 @@ func (params ConsensusParams) ValidateBasic() error {
 		return fmt.Errorf("Feature.PbtsEnableHeight cannot be negative. Got: %d", params.Feature.PbtsEnableHeight)
 	}
 
+	// FIXME: should we enforce something more than > 0? Maybe >= time.Millisecond?
```
Comment (sergio-mena) [inline types/params.go:254]: I would be in favor of that, for practical reasons. @lasarojc ? @glnro ?

File: types/params_test.go [hunk 3]
```diff
@@ -209,27 +209,27 @@ func TestConsensusParamsValidation(t *testing.T) {
...
+		// test invalid synchrony params
 		{
-			name: "messageDelay -1",
+			name: "messageDelay 0",
 			params: makeParams(makeParamsArgs{
 				evidenceAge:  2,
-				precision:    1,
-				messageDelay: -1,
+				precision:    time.Nanosecond,
+				messageDelay: 0,
```
Comment (sergio-mena) [inline types/params_test.go:223]: Why this change?

File: types/params_test.go [hunk 4]
```diff
@@ -209,27 +209,27 @@ func TestConsensusParamsValidation(t *testing.T) {
...
 			valid: false,
 		},
 		{
-			name: "precision -1",
+			name: "precision 0",
 			params: makeParams(makeParamsArgs{
 				evidenceAge:  2,
-				precision:    -1,
-				messageDelay: 1,
+				precision:    0,
```
Comment (sergio-mena) [inline types/params_test.go:231]: And this one?

File: internal/consensus/msgs_test.go [hunk 5]
```diff
@@ -50,10 +50,10 @@ func TestMsgToProto(t *testing.T) {
 	proposal := types.Proposal{
 		Type:      types.ProposalType,
 		Height:    1,
-		Round:     1,
+		Round:     2,
```
Comment (lasarojc) [inline internal/consensus/msgs_test.go:53]: Any particular reason for this change, or just shaking things up?

File: types/params.go (hunk 2 above)
Comment (lasarojc) [reply types/params.go:254]: I don't see the reason. If it is large, may be too large. And if it is small, like `time.Millisecond`, then it is just as good as checking `<= 0` imho.
Unless having an increment of less than 1ms could problematic, which I don't think it is since timestamps have ns resolution.

File: types/params_test.go (hunk 4 above)
Comment (lasarojc) [reply types/params_test.go:231]: Anything <= 0 is invalid, so this is the largest value that already triggers the failure.

Comment (cason) [reply types/params_test.go:231]: 0 should be invalid

File: types/params_test.go (hunk 3 above)
Comment (cason) [reply types/params_test.go:223]: 0 should be invalid

File: internal/consensus/msgs_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (cason) [reply internal/consensus/msgs_test.go:53]: POLRound < Round.

File: types/canonical.go (hunk 1 above)
Comment (cason) [reply types/canonical.go:47]: I can remove, but the proto name does not match the type name.

File: types/params_test.go [hunk 6]
```diff
@@ -252,8 +252,8 @@ func TestConsensusParamsValidation(t *testing.T) {
 				makeParamsArgs{
 					blockBytes:   1,
 					evidenceAge:  2,
-					precision:    1,
-					messageDelay: 1,
+					precision:    time.Nanosecond,
+					messageDelay: time.Nanosecond,
 					pbtsHeight:   0,
 				}),
 			valid: true,
```
[the remark this answers is not in the document]
Comment (cason) [reply types/params_test.go:259]: What is the difference?

File: types/params_test.go [hunk 7]
```diff
@@ -654,3 +654,36 @@ func TestProtoUpgrade(t *testing.T) {
...
+	lastSP := originalSP
+	for round := int32(1); round <= 10; round++ {
+		adaptedSP := adaptiveSP(originalSP, round)
+		assert.NotEqual(t, adaptedSP, lastSP)
+		assert.Equal(t, adaptedSP.Precision, lastSP.Precision,
+			"Precision must not change over rounds")
+		assert.Greater(t, adaptedSP.MessageDelay, lastSP.MessageDelay,
+			"MessageDelay must increase over rounds")
+
+		// It should not increase a lot per round, say more than 25%
```
[the remark this answers is not in the document]
Comment (cason) [reply types/params_test.go:677]: It can be. I didn't want to be super precise here, we only need it to increase.

File: types/canonical.go (hunk 1 above)
Comment (sergio-mena) [reply types/canonical.go:47]: You mean `POLRound` vs `PolRound` ?

Comment (cason) [reply types/canonical.go:47]: Yep

File: types/params.go (hunk 2 above)
Comment (sergio-mena) [reply types/params.go:254]: OK, so let's keep it simple. Let's leave the `>0`

Comment (cason) [reply types/params.go:254]: Up to you.

File: types/params_test.go (hunk 3 above)
Comment (sergio-mena) [reply types/params_test.go:223]: Then, can you test both, 0 and -1? (i.e., add an extra TC here)

File: types/params_test.go (hunk 4 above)
Comment (sergio-mena) [reply types/params_test.go:231]: Same, let's test both, in case someone, 3 years from now, decides it should be `<0` in production

File: types/params_test.go (hunk 3 above)
Comment (cason) [reply types/params_test.go:223]: Ok.

File: types/params.go (hunk 2 above)
Comment (cason) [reply types/params.go:254]: So remove the FIXME?

File: types/params_test.go (hunk 3 above)
Comment (sergio-mena) [reply types/params_test.go:223]: 🙏

File: types/canonical.go (hunk 1 above)
Comment (sergio-mena) [reply types/canonical.go:47]: Added a bullet point in #1731

**Review by lasarojc — COMMENTED** [review]
> What is the difference? 

PBTS is enabled

**Review by lasarojc — APPROVED** [review]

**Review by sergio-mena — APPROVED** [review]
